### PR TITLE
ROC-5753 fix states paid rejected dashboard

### PR DESCRIPTION
**Pull request by andrewwa-kainos**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-5753

### Change description
Update rejected states paid dashboards to distinguish between mediation and no mediation cases

### Work checklist

- [ ] Unit tests added where applicable
- [x] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [x] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/features/dashboard/views/macro/claimStatus/statesPaidRejected.njk [hunk 1]
```diff
@@ -1,25 +1,39 @@
...
+  <p>{{ t('You’ve both agreed to try mediation. The Small Claims Mediation Service will contact you to arrange an appointment.') }}</p>
+  {{ externalLink('Find out how mediation works', 'https://www.gov.uk/government/publications/small-claims-mediation-service-ex730') }}
+{% endmacro %}
 
-  <p>{{ _downloadResponseLink(claim, 'Download their response') }} {{ t('(PDF)') }}</p>
+{% macro statesPaidRejectedMediationNOClaimantClaimDetails(claim) %}
+  <h2 class="heading-small">{{ t('Wait for the court to review the case') }}</h2>
+  <p>{{ t('You’ve rejected {{ defendantName }}’s response and said you want to take the case to court.',
+      { defendantName: claim.claimData.defendant.name }) }}</p>
+  <p>{{ t('The court will review the case. We’ll email you if we set a hearing date to tell you how to prepare.') }}</p>
```
Comment (robertparkinson) [inline src/main/features/dashboard/views/macro/claimStatus/statesPaidRejected.njk]: this should be the paper DQs version of this status. this one assumes that DQs have been turned on

File: src/main/features/dashboard/views/macro/claimStatus/statesPaidRejected.njk [hunk 2]
```diff
@@ -40,19 +54,33 @@
...
+
   <p>{{ downloadResponseLink(claim) }}</p>
 {% endmacro %}
 
+{% macro statesPaidRejectedMediationYESDefendantClaimDetails(claim) %}
+  <p>{{ t('They agreed to try mediation. We’ll contact you with details of your appointment.') }}</p>
+{% endmacro %}
+
+{% macro statesPaidRejectedMediationNODefendantClaimDetails(claim) %}
+  <p>{{ t('You might have to go to a court hearing. We’ll contact you if we set a hearing date to tell you how to prepare.') }}</p>
```
Comment (robertparkinson) [inline src/main/features/dashboard/views/macro/claimStatus/statesPaidRejected.njk]: as above, this should say something about completing the paper DQs

File: src/main/features/dashboard/views/macro/claimStatus/statesPaidRejected.njk (hunk 1 above)
Comment (andrewwa-kainos) [reply src/main/features/dashboard/views/macro/claimStatus/statesPaidRejected.njk]: DQs content is being reviewed by Steve and I've been told to progress as is for now

File: src/main/features/dashboard/views/macro/claimStatus/statesPaidRejected.njk (hunk 2 above)
Comment (andrewwa-kainos) [reply src/main/features/dashboard/views/macro/claimStatus/statesPaidRejected.njk]: DQs content is being reviewed by Steve and I've been told to progress as is for now

File: src/main/features/dashboard/views/macro/claimStatus/statesPaidRejected.njk [hunk 3]
```diff
@@ -40,19 +54,34 @@
...
-  <h2 class="heading-small">{{ t('{{ claimantName }} rejected your admission of {{ amount }}', {
-      claimantName: claim.claimData.claimant.name,
-      amount: amount | numeral
-    }) }}</h2>
+  {% set mediation = claim.claimantResponse.freeMediation and claim.claimantResponse.freeMediation === YesNoOption.YES.option %}
+
+  {% if mediation %}
+    <h2 class="heading-small">{{ t('We’ll contact you with a mediation appointment') }}</h2>
+  {% else %}
+    <h2 class="heading-small">{{ t('The claimant has rejected your admission of {{ amount }}', { amount: amount | numeral}) }}</h2>
```
Comment (srineelesh) [inline src/main/features/dashboard/views/macro/claimStatus/statesPaidRejected.njk]: I think we need to display the claimant's name here according to the prototype

File: src/main/features/dashboard/views/macro/claimStatus/statesPaidRejected.njk [hunk 4]
```diff
@@ -40,19 +54,34 @@
...
+    {{ statesPaidRejectedMediationYESDefendantClaimDetails(claim) }}
+  {% else %}
+    {{ statesPaidRejectedMediationNODefendantClaimDetails(claim) }}
+  {% endif %}
+
   <p>{{ downloadResponseLink(claim) }}</p>
 {% endmacro %}
 
+{% macro statesPaidRejectedMediationYESDefendantClaimDetails(claim) %}
+  <p>{{ t('They agreed to try mediation. We’ll contact you with details of your appointment.') }}</p>
```
Comment (srineelesh) [inline src/main/features/dashboard/views/macro/claimStatus/statesPaidRejected.njk]: minor: in the prototype it says `They've`

File: src/test/features/dashboard/routes/claimStatus/full-defence.ts [hunk 5]
```diff
@@ -74,15 +73,15 @@ const testData = [
       ...directionsQuestionnaireDeadline
     },
     claimantAssertions: [
-      'You’ve rejected the defendant’s admission',
-      `They said they owe ${NumberFormatter.formatMoney(defenceWithAmountClaimedAlreadyPaidData.paymentDeclaration.paidAmount)}`,
-      'complete a directions questionnaire',
+      'Wait for the court to review the case',
```
Comment (srineelesh) [inline src/test/features/dashboard/routes/claimStatus/full-defence.ts]: question: do we have any tests for the new content ?

File: src/main/features/dashboard/views/macro/claimStatus/statesPaidRejected.njk (hunk 4 above)
Comment (andrewwa-kainos) [reply src/main/features/dashboard/views/macro/claimStatus/statesPaidRejected.njk]: Agreed it's minor - have highlighted the inconsistency to UX team

File: src/main/features/dashboard/views/macro/claimStatus/statesPaidRejected.njk (hunk 3 above)
Comment (andrewwa-kainos) [reply src/main/features/dashboard/views/macro/claimStatus/statesPaidRejected.njk]: As discussed, this matches the January prototype I was instructed to follow. UX should be updating the FTBB prototype accordingly.

File: src/test/features/dashboard/routes/claimStatus/full-defence.ts (hunk 5 above)
Comment (andrewwa-kainos) [reply src/test/features/dashboard/routes/claimStatus/full-defence.ts]: Added tests for mediation agreed case

File: src/main/features/dashboard/views/macro/claimStatus/statesPaidRejected.njk (hunk 4 above)
Comment (andrewwa-kainos) [reply src/main/features/dashboard/views/macro/claimStatus/statesPaidRejected.njk]: Mistook the above as a shared component - this should be changed.

**Pull request merged**